uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/manage-leave-requests.component.js

```diff
@@ -254,36 +254,43 @@ define([
     }
 
     /**
-     * Loads the managees and calls loadLeaveRequests()
+     * Loads the leave requests for the given data type
      *
+     * @param {String} dataType - which data to load, table or filter or both
      * @return {Promise}
      */
-    function loadManageesAndLeaves () {
-      vm.loading.content = true;
+    function loadData (dataType) {
```

Comment:
and actually, after taking a look at the code a bit more, why do you need this function at all? It's gonna confused things more. You definitely need to rename this to something like `loadLeaveRequestsXYZ`, but then you already have a `loadLeaveRequests` function that also accepts a "section" param! There is some duplication here and it's definitely not needed.

Since you are calling `loadData` only inside `loadManageesAndLeaves`, why don't you simply put this logic in that method and get rid of this? Granted, it would put back a bit too much "meat" in the method, but i think it's better than how things are now in this pr